### PR TITLE
highlight property override change

### PR DESCRIPTION
**Pull request by kunkun-tang**
The AZ production users complained that 

> We've had a number of production issues where config properties had been
overridden in the Azkaban UI as a hot fix but then those changes didn't get
reflected in git. Subsequent deployments overrode those changes which
caused the issues.

To resolve this issue, the first step discussed is to highlight the property override change in project log.

Added Unit Test and Evaluated in my lcoalhost environment.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/project/ProjectManager.java [hunk 1]
```diff
@@ -354,16 +353,22 @@ public Props getJobOverrideProperty(Project project, String jobName)
     return projectLoader.fetchProjectProperty(project, jobName + ".jor");
   }
 
-  public void setJobOverrideProperty(Project project, Props prop, String jobName)
+  public void setJobOverrideProperty(Project project, Props prop, String jobName, User modifier)
       throws ProjectManagerException {
     prop.setSource(jobName + ".jor");
```
Comment (chengren311) [inline azkaban-common/src/main/java/azkaban/project/ProjectManager.java]: can we put this ".jor" to constants files btw ?

Comment (kunkun-tang) [inline azkaban-common/src/main/java/azkaban/project/ProjectManager.java]: What's the advantage of this change? I can't find many usages of ".jor".

File: azkaban-common/src/test/java/azkaban/utils/PropsUtilsTest.java [hunk 2]
```diff
@@ -208,6 +208,30 @@ public void testCyclesResolveProps() throws IOException {
...
+
+    newProps2.put("a", "a_value1");
+    newProps2.put("b", "b_value1");
+    newProps2.put("c", "c_value1");
+
+    String message2= PropsUtils.getPropertyDiff(oldProps, newProps2);
+
+    Assert.assertEquals(message2, "Newly created Properties: [ c, c_value1], \n");
+  }
+
```
Comment (jamiesjc) [inline azkaban-common/src/test/java/azkaban/utils/PropsUtilsTest.java]: Do we need to test diffProperties?

Comment (kunkun-tang) [inline azkaban-common/src/test/java/azkaban/utils/PropsUtilsTest.java]: Thanks @jamiesjc for reminding me adding an extreme test case.

File: azkaban-common/src/main/java/azkaban/project/ProjectManager.java (hunk 1 above)
Comment (shuzhang1989) [inline azkaban-common/src/main/java/azkaban/project/ProjectManager.java]: @chengren311 +1, don't see the advantage

**Pull request merged**